File: tests/fixtures/go/syntax_errors.go
package fixtures

// ValidHeaderFunc is well-formed before any syntax breaks.
func ValidHeaderFunc(a, b int) int {
	return a + b
}

// BrokenBraceFunc has unclosed brackets and malformed control flow.
func BrokenBraceFunc(items []string) string {
	for _, it := range items {
		if len(it) > 0 {
			return it
	// Missing closing braces for if and for

// TargetFuncWithSurroundingErrors is an intact function surrounded by syntax errors.
func TargetFuncWithSurroundingErrors(val int) int {
	return val*val + 100
}

// MalformedStruct has missing field types and invalid syntax tokens.
type MalformedStruct struct {
	FieldA string
	FieldB 
	FieldC [][]int
	invalid token %%%% here

File: tests/fixtures/rust/syntax_errors.rs
//! Rust syntax error fixture for compiler and tree-sitter error recovery tests.

/// Intact header function before any syntax faults.
pub fn valid_header_function(x: i32, y: i32) -> i32 {
    x + y
}

/// Function with unclosed blocks and missing braces.
pub fn broken_braces_fn(items: Vec<String>) -> Vec<String> {
    for item in items {
        if item.len() > 0 {
            println!("Item: {}", item;
    // Missing parenthesis, semicolon, and 2 closing braces

/// Intact target function embedded inside a syntactically invalid file.
pub fn target_intact_function<'a>(input: &'a str) -> &'a str {
    input.trim()
}

/// Unbalanced macro invocation.
macro_broken!(
    struct MalformedStruct {
        field_a: i32,
        invalid lifetime syntax '123_invalid,
